themes/default/content/blog/serverless-architecture-design/index.md

```diff
@@ -0,0 +1,115 @@
+---
+title: "Serverless Architecture Design with Pulumi"
+date: 2023-10-30T20:40:47Z
+draft: false
+meta_desc: "Learn about the concept of serverless architecture and how to use Pulumi and Go to create your Azure serverless application"
+meta_image: meta.png
+authors:
+    - adora-nwodo
+tags:
+    - serverless
+
+---
+
+Serverless architectures have gained more popularity in the recent years because they make it easier for software developers to focus on application development without the additional operation overhead. With Pulumi, you can deploy and manage your serverless infrastructure and this article shows you how to do this with Azure.
+
+Before we go into the specifics of serverless architecture design with Pulumi on azure, let's go
+through the basics of serverless - this is important for the rest of the article.
+
+## What is Serverless Architecture
+
+Serverless architecture is a cloud computing model that simplifies the management of infrastructure. This allows software developers to focus on writing code without worrying
+about server provisioning, scaling or maintenance. The idea behind it is functions as a service (FaaS), in which the environment dynamically scales in response to demand and code is executed in response to events.
+
+Key characteristics of serverless architecture include:
+
+- Event-Driven: Serverless functions are triggered by events in the system. These events could be HTTP requests, messages in a queue, or changes to a data store.
+
+- Auto-Scaling: With serverless, resources automatically scale up or down to match the workload so you don't use infrastructure beyond what you actually need.
+
+- Pay-as-You-Go: You are charged only for the actual compute resources you use during code execution.
+
+## Requirements
+
+To follow along with this article, here are some things you might need:
+
+- [Pulumi CLI](https://www.pulumi.com/docs/install/)
+
+- [Pulumi account and access token](https://app.pulumi.com/signup)
+
+- [Go](https://go.dev/doc/install)
+
+- [Git CLI](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+
+- [Az CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
+
+Now that you have everything needed to get started, let's see our serverless application written in Go and deployed using Pulumi.
+
+### 1. Clone the repo and setup your project
```

Comment:
we are missing an entire section talking about what we are building here. we are just somehow jumping into doing an example. the reader has no idea what they are actually building when following your instructions. 

we should have an arch diagram. we should have and explanation of all the pieces. 

we are also missing an entire section about why any of these stuff is hard and how does Pulumi help. why should they use Pulumi versus whatever their previous workflow is?